protocol-tokio/src/protocol/inbound_stream.rs

```diff
@@ -74,8 +80,8 @@ impl<T: AsyncRead, B: property::Block + property::HasHeader, Tx: property::Trans
 where
     B: cbor_event::Deserialize,
     B: cbor_event::Serialize,
-    B::Id: cbor_event::Deserialize,
-    B::Id: cbor_event::Serialize,
+    <B as property::Block>::Id: cbor_event::Deserialize,
+    <B as property::Block>::Id: cbor_event::Serialize,
```

Comment:
Might as well put both bounds on one line joined with a `+`, here and in other places below.

Reply:
Are these changes a remnant of having the `Id` associated type on `HasHeader`? This disambiguation should not be needed.